src/macros.rs

```diff
@@ -33,7 +33,35 @@ macro_rules! handle_command {
             if command.is_ansi_code_supported() {
                 write_ansi_code!($writer, command.ansi_code())
             } else {
-                command.execute_winapi().map_err($crate::ErrorKind::from)
+                command
+                    .execute_winapi(&mut $writer)
+                    .map_err($crate::ErrorKind::from)
+            }
+        }
+        #[cfg(unix)]
+        {
+            write_ansi_code!($writer, $command.ansi_code())
+        }
+    }};
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! handle_fmt_command {
+    ($writer:expr, $command:expr) => {{
+        use $crate::{write_ansi_code, Command};
+
+        #[cfg(windows)]
+        {
+            let command = $command;
+            if command.is_ansi_code_supported() {
+                write_ansi_code!($writer, command.ansi_code())
+            } else {
+                command
+                    // $writer is not used in execute_winapi when called from handle_fmt_command macro
+                    // we give it stdout() as place holder to satisfy the type requirement
+                    .execute_winapi(&mut std::io::stdout())
```

Comment:
Can we use something like this:

```
struct FakeStdout;

impl Write for FakeStdout {
    fn write(&mut self, buf: &[u8]) -> Result<usize> {
        unimplemented!("This is a fake stdout and is not used anywhere")
    }

    fn flush(&mut self) -> Result<()> {
        unimplemented!("This is a fake stdout and is not used anywhere")
    }
}
```

The problem if we are using `&mut std::io::stdout()` is that it will request a write unlock for stdout which is unnecessary. We can also use a fake `Write` implementation.

Reply:
I like this so much! (especially the guarantee we get from unimplemented!).
I just changed the name to FakeWriter, I think its more correct.